### PR TITLE
Fix of incorrect wrap position in wrap_rows

### PR DESCRIPTION
**Pull request by figec**
Commit 43dab2ffc899133fbd8bb4bf4d209a3bbc70b901 fixes UTF-8 string length calculation in wrap_rows, but newline inserted after first byte of char and destroy multibyte chars.

Example of incorrect behavior:
```
2015-06-18 20:33:20: INFO[main]: Couldn't convert UTF-8 string 0x4d696e65746573740a5645523d302e342e31322d6465762d65343565636164204255494c445f545950453d44656275672052554e5f494e5f504c4143453d31205553455f474554544558543d30205553455f534f554e443d31205553455f4355524c3d31205553455f46524545545950453d31205553455f4c55414a49543d31205354415449435f53484152454449523d222e220a706174685f75736572203d202f686f6d652f726170746f722fd0a0d0b0d0b7d180d0b0d0b1d00abe0ad182d0bad0b02f6370702f6d696e65746573742f62696e0a2f2e2e0a into wstring
```

Also fixed high bound in IS_UTF8_MULTB_INNER macro.

### CODE REVIEW

File: src/util/string.h [hunk 1]
```diff
@@ -426,12 +426,20 @@ inline std::string wrap_rows(const std::string &from,
...
+	bool need_to_wrap = false;
+
 	size_t character_idx = 0;
 	for (size_t i = 0; i < from.size(); i++) {
 		if (character_idx > 0 && character_idx % row_len == 0)
-			to += '\n';
-		if (!IS_UTF8_MULTB_INNER(from[i]))
+			need_to_wrap = true;
+		if (!IS_UTF8_MULTB_INNER(from[i])) {
+			// Warp string if needed before next char started
```
Comment (est31) [inline src/util/string.h]: Wrap not warp :)

**Comment by est31** [comment]
Can you add a unittest? E.g. do it like in 43dab2ffc899133fbd8bb4bf4d209a3bbc70b901 but with data `2f686f6d652f726170746f722fd0a0d0b0d0b7d180d0b0d0b1d00abe0ad182d0bad0b02f6370702f6d696e65746573742f62696e0a2f2e2e0a`

**Comment by figec** [comment]
> Wrap not warp :)

Fixed.

**Comment by figec** [comment]
> Can you add a unittest? E.g. do it like in 43dab2f but with data 2f686f6d652f726170746f722fd0a0d0b0d0b7d180d0b0d0b1d00abe0ad182d0bad0b02f6370702f6d696e65746573742f62696e0a2f2e2e0a

Need to replace s[] with new data, or create new variable?

**Comment by est31** [comment]
As you like :)

**Comment by est31** [comment]
I wonder does only moving the increment work too? If yes this would be simpler, and we still have "insert before char".
```
	std::string to;

	size_t character_idx = 0;
	for (size_t i = 0; i < from.size(); i++) {
		if (!IS_UTF8_MULTB_INNER(from[i]))
			character_idx++;
		if (character_idx > 1 && (character_idx - 1) % row_len == 0)
			to += '\n';
		to += from[i];
	}

	return to;
```

**Comment by figec** [comment]
> I wonder does only moving the increment work too?

No. For example: `S = single byte char, B = begin byte, I = inner byte.`

If we need to wrap string structured as `SSBIBIBII` by 3. At byte number 3 we will make character_idx = 3 and then insert newline after this position. As result we will have a broken string at output.

**Comment by figec** [comment]
testWrapRows() data changed.

**Comment by est31** [comment]
Thanks for the fix 6044d413fedf571e2444fdd8b8a5b5be0755ff3f

**Comment by est31** [comment]
Sorry have misunderstood you with change s[] vs new variable. The more test cases, the better :)

**Pull request closed without merging**